src/main/java/com/seals/camplanner/event/controllers/EventController.java

```diff
@@ -12,25 +15,44 @@
 @RequiredArgsConstructor
 public class EventController {
     private final EventService eventService;
+    private final LocationService locationService;
+    private final ModelMapper modelMapper;
 
     @GetMapping("/event")
-    public List<Event> getEvents() {
-        return eventService.findAll();
+    public List<EventDto> getEvents() {
+        return toEventDtoList(eventService.findAll());
     }
 
     //TODO add treatment for the "location" parameter
     @PostMapping("/event")
-    public Event saveEvent(@RequestBody EventDto event) {
-        return eventService.save(event);
+    public EventDto saveEvent(@RequestBody EventDto event) {
+        return toDto(eventService.save(fromDto(event)));
     }
 
     @GetMapping("/event/{id}")
-    public Event getEvent(@PathVariable("id") Long id) {
-        return eventService.find(id);
+    public EventDto getEvent(@PathVariable("id") Long id) {
+        return toDto(eventService.find(id));
     }
 
     @DeleteMapping("/event/{id}")
     public void deleteEvent(@PathVariable("id") Long id) {
         eventService.delete(id);
     }
+
+    private EventDto toDto(Event event) {
+        EventDto eventDto = modelMapper.map(event, EventDto.class);
+        eventDto.setLocationId(event.getLocation().getId());
```

Comment:
I ran the `shouldMatchValuesMappingEventEventDto` (after fixing it), and it properly set the ID without having to manually do it. This line might not be necessary.